src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeParse.cs

```diff
@@ -4055,7 +4057,7 @@ private static bool ParseByFormat(
                         {
                             parseInfo.fUseTwoDigitYear = true;
                         }
-                        parseResult = ParseDigits(ref str, tokenLen, out tempYear);
+                        parseResult = ParseDigits(ref str, tokenLen, maxYearTokenLength, out tempYear);
                     }
```

Comment:
parseResult = ParseDigits(ref str, tokenLen, maxDigitLen: tokenLen <= 2 ? 2 : 4, out tempYear);

Reply:
I think the test https://helixr1107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-115507-merge-c1d147295e6d448da4/System.Net.Http.Functional.Tests/3/console.a9f6f133.log?helixlogtype=result failed because you didn't handle the token length `2`. Try the suggestion and let's see if this test will still fail.